python-package/xgboost/core.py

```diff
@@ -1193,6 +1193,7 @@ def __init__(self, params=None, cache=(), model_file=None):
 
         params = params or {}
         params = self._configure_metrics(params.copy())
+        params = self._configure_constraints(params)
```

Comment:
It might be a good idea to make a copy since you are modifying the value.  Python arguments are passed as reference so after training the parameter in users' code will be changed.

Reply:
NVM.  Just saw the copy.